docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst

```diff
@@ -114,10 +129,19 @@ The models are stored in a series of YAML data extension files, one for each pub
 
 The editor will create a separate model file for each package that you model.
 
-Testing CodeQL model packs
---------------------------
+Modeling methods that can be called in different ways
```

Comment:
@starcke I think you might be able to explain this better than me. It is that a method is callable in different ways that means we want multiple classifications? I'm not sure that sounds right (but I could be wrong).

Reply:
Each classification (or model) refers to a single flow (e.g. parameter 1 might be a sql-injection sink). Each method might have multiple flows. For example, consider a method that executes a query and returns it:
```
String runQuery(String query)
```
Here `q` is likely to be a sink for SQL injection. But at the same time because the query is also returned it is also a flow summery where taint can flow from the first parameter to the return value. This could potentially result in other vulnerabilities (e.g. a log injection vulnerability if the query is later logged).

Reply:
The reason we need multiple classifications/models is to capture all the possible flows for a given method.

Reply:
Thanks for this discussion. I've updated the title and introduction to this section accordingly.